Xamarin.Forms.Build.Tasks/XmlTypeExtensions.cs

```diff
@@ -3,30 +3,49 @@
 using System.Xml;
 using Mono.Cecil;
 using Mono.Cecil.Rocks;
-using Xamarin.Forms.Internals;
 using Xamarin.Forms.Xaml;
+using Xamarin.Forms.Internals;
 
 namespace Xamarin.Forms.Build.Tasks
 {
 	static class XmlTypeExtensions
 	{
-		static IList<XmlnsDefinitionAttribute> s_xmlnsDefinitions;
+		static Dictionary<ModuleDefinition, IList<XmlnsDefinitionAttribute>> s_xmlnsDefinitions = 
+			new Dictionary<ModuleDefinition, IList<XmlnsDefinitionAttribute>>();
+		static object _nsLock = new object();
 
-		static void GatherXmlnsDefinitionAttributes()
+		static IList<XmlnsDefinitionAttribute> GatherXmlnsDefinitionAttributes(ModuleDefinition module)
 		{
-			//this could be extended to look for [XmlnsDefinition] in all assemblies
-			var assemblies = new [] {
-				typeof(XamlLoader).Assembly,
-				typeof(View).Assembly,
-			};
-
-			s_xmlnsDefinitions = new List<XmlnsDefinitionAttribute>();
-
-			foreach (var assembly in assemblies)
-				foreach (XmlnsDefinitionAttribute attribute in assembly.GetCustomAttributes(typeof(XmlnsDefinitionAttribute), false)) {
-					s_xmlnsDefinitions.Add(attribute);
-					attribute.AssemblyName = attribute.AssemblyName ?? assembly.FullName;
+			var xmlnsDefinitions = new List<XmlnsDefinitionAttribute>();
+
+			if (module.AssemblyReferences?.Count > 0) {
+				// Search for the attribute in the assemblies being
+				// referenced.
+				foreach (var asmRef in module.AssemblyReferences) {
+					var asmDef = module.AssemblyResolver.Resolve(asmRef);
+					foreach (var ca in asmDef.CustomAttributes) {
+						if (ca.AttributeType.FullName == typeof(XmlnsDefinitionAttribute).FullName) {
+							var attr = GetXmlnsDefinition(ca, asmDef);
+							xmlnsDefinitions.Add(attr);
+						}
+					}
 				}
+			} else {
+				// Use standard XF assemblies
+				// (Should only happen in unit tests)
+				var requiredAssemblies = new[] {
```

Comment:
I don't think this would ever happen in real life but this method was getting indirectly called in one of the unit tests with a dummy `module` that was assuming the standard assemblies were referenced.